### PR TITLE
Implement SPV_INTEL_optnone extension

### PR DESCRIPTION
**Pull request by NikitaRudenkoIntel**
Spec: https://github.com/intel/llvm/pull/3198

### CODE REVIEW

File: include/spirv/unified1/spirv.core.grammar.json
Comment (AlexeySachkov) [inline include/spirv/unified1/spirv.core.grammar.json:13461]: Strictly speaking, the capability is called `CapabilityOptNoneINTEL`. I'm not familiar enough with SPIR-V Headers internals to say whether it is a good name or not, because presumably we will get `CapabilityOptNoneINTEL` enumerant in `Capability` enum, which seems repetitive.

Tagging @Fznamznon and @bashbaug here to comment

Comment (Fznamznon) [reply include/spirv/unified1/spirv.core.grammar.json:13461]: I'm not sure that calling this capability `CapabilityOptNoneINTEL` was a good idea, so that is probably just a bug in spec and it should be called `OptNoneINTEL` instead.

Comment (NikitaRudenkoIntel) [reply include/spirv/unified1/spirv.core.grammar.json:13461]: Can I merge this PR?

Comment (AlexeySachkov) [reply include/spirv/unified1/spirv.core.grammar.json:13461]: > I'm not sure that calling this capability CapabilityOptNoneINTEL was a good idea, so that is probably just a bug in spec and it should be called OptNoneINTEL instead.

Won't it create an ambiguity for the spec, i.e. when you use `OptNoneINTEL`, which particular thing you refer to? Is it Capability or is it Declaration? However, I guess that we can always explicitly state which exact thing we refer to.

@NikitaRudenkoIntel, I think we can go ahead and merge this PR and adjust the spec. @Fznamznon, will you take care about the spec changes?

**Review by AlexeySachkov — APPROVED** [review]
I'm not really familiar with the repo and the process of modifying `.json` grammar files, but naming and opcodes together with header file changes seems to be correct.

I'm also unfamiliar with the review/approve processes in this repo, so I don't know if approve for me is enough to get this merged

File: include/spirv/unified1/spirv.core.grammar.json
Comment (mmerecki) [inline include/spirv/unified1/spirv.core.grammar.json:8695]: The following fields are missing:
```yaml
"capabilities" : [  "OptNoneINTEL" ],
"version" : "None"
```

**Comment by bashbaug** [comment]
Hi, for pull requests like these we should wait for approval by a non-Intel reviewer before merging.  Open PRs are usually reviewed and merged in the SPIR-V working group meeting.  In the meantime, these changes should be reverted.

Please also see the comments from @mmerecki that need to be addressed before merging again.

**Pull request merged**
